contracts/minter/src/contract.rs

```diff
@@ -524,10 +543,17 @@ pub fn execute_update_batch_mint_limit(
     Ok(Response::new().add_attribute("action", "update_batch_mint_limit"))
 }
 
-pub fn mint_price(deps: Deps) -> Result<Coin, StdError> {
-    // if in whitelist, use whitelist price. else use config unit price
+pub fn mint_price(deps: Deps, admin_no_fee: bool) -> Result<Coin, StdError> {
+    // if admin => admin mint fee,
+    // else if in whitelist => whitelist price
+    // else => config unit price
     let config = CONFIG.load(deps.storage)?;
-    let mint_price: Coin = if let Some(whitelist) = config.whitelist {
+    let mint_price: Coin = if admin_no_fee {
```

Comment:
Instead of this, you can also make the `mint_price` an `Option` type right?

Reply:
i.e: The function signature can probably just be `fn mint_price(deps: Deps, admin_no_fee: bool) -> Option<Coin>`